src/box/engine.c

```diff
@@ -228,7 +228,13 @@ engine_prepare_join(struct engine_join_ctx *ctx)
 int
 engine_join(struct engine_join_ctx *ctx, struct xstream *stream)
 {
-	ERROR_INJECT_YIELD(ERRINJ_ENGINE_JOIN_DELAY);
+	ERROR_INJECT_WHILE(ERRINJ_ENGINE_JOIN_DELAY, {
+		fiber_sleep(0.001);
+		if (fiber_is_cancelled()) {
+			diag_set(FiberIsCancelled);
+			return -1;
+		}
+	});
```

Comment:
Do we really need to care about this? Tests shouldn't leave the error injection set at exit. Did you encounter any problems because of this injection?